pywwt/annotation.py

```diff
@@ -252,6 +255,100 @@ def _on_trait_change(self, changed):
+        self.dim = self.parent.instruments.dim
+        
+        # list of IDs of annotations created in this FieldofView instance
+        self.active = []
+        self._gen_fov(telescope, center, rot, **kwargs)
+
+    def _gen_fov(self, telescope, center, rot, **kwargs):
+        #telescope = telescope.lower()
+        if telescope not in self.available:
+            raise ValueError('the given telescope\'s field of view is unavailable at this time')
+
+        position = self.dim[telescope][0]
+        dimensions = self.dim[telescope][-1]
+        # dimensions is a list of lists [ [[ras], [decs]], '', ...]
+
+        # test that pos matches what the user entered
+        if center and position == 'absolute':
+            raise ValueError('the given telescope does not take center coordinates')
+        elif not center and position == 'relative':
+            raise ValueError('the given telescope requires center coordinates')
+
+        # draw jwst nircam's inner panels first
+        if telescope[-6:] == 'nircam':
+            self._gen_fov('nircam_short', center, rot, **kwargs)
+
+        for panel in dimensions:
+            ras = (panel[0] * u.deg).value
+            decs = (panel[1] * u.deg).value
+            ra_r, dec_r = self._rotate(ras, decs, rot)
+
+            if telescope == 'k2':
+                ra_tr = ra_r; dec_tr = dec_r
+            else:
+                center_ra = center.ra.to(u.deg).value
+                center_dec = center.dec.to(u.deg).value
+
+                dec_tr = [center_dec + dec for dec in dec_r]
+                # scale RA by dec (due to polar contraction of spherical coords)
+                ra_tr = [center_ra +
+                         ra / np.cos((dec_tr[i] * u.deg).to(u.rad)).value
+                         for i, ra in enumerate(ra_r)]
+
+                # check that abs(dec) < 90. if not, adjust it, and then
+                # set the corresponding ra to be swapped by 180
+                for i, dec in enumerate(dec_tr):
+                    if abs(dec) > 90:
+                        if dec < -90:
+                            dec_tr[i] = -90. - (dec + 90.)
+                        else: # dec > 90
+                            dec_tr[i] = 90. - (dec - 90.)
+                        ra_tr[i] += 180.
+
+            corners = SkyCoord(ra_tr, dec_tr, unit='deg')
+            if self.parent.galactic_mode:
+                corners = corners.galactic
+
+            #print(corners)
+            annot = self.parent.add_polygon(corners, **kwargs)
+            self.active.append(annot)
+
+    def _rotate(self, ras, decs, rot):
+        # now, we rotate shapes at the origin before translating
+        rot = rot.to(u.rad).value
+        cos = np.cos(rot); sin = np.sin(rot)
+        ra_rot = []; dec_rot = []
+
+        i = 0
+        while i < len(ras):
+            ra_rot.append(ras[i]*cos - decs[i]*sin)
+            dec_rot.append(ras[i]*sin + decs[i]*cos)
+            i += 1
```

Comment:
I don't think the loop should be needed? That is, I think the following should work:

```python
ra_rot = ras * cos - decs * sin
dec_rot = ras * sin + decs * cos
```

Reply:
I have the same comment on using lists here as on lines 306 and 308. 

I considered using `np.array` so I wouldn't need a loop, but the while loop ended up being 2-5 times faster when I measured speeds with `%timeit`. This makes a difference with big footprints like K2, which takes about a minute to draw right now.